core/src/main/scala/kafka/server/RaftServer.scala

```diff
@@ -47,13 +47,18 @@ class RaftServer(val config: KafkaConfig) extends Logging {
 
   private val partition = new TopicPartition("__cluster_metadata", 0)
   private val time = Time.SYSTEM
+  private val shutdownLatch = new CountDownLatch(1)
 
   var socketServer: SocketServer = _
   var credentialProvider: CredentialProvider = _
   var tokenCache: DelegationTokenCache = _
   var dataPlaneRequestHandlerPool: KafkaRequestHandlerPool = _
   var scheduler: KafkaScheduler = _
   var metrics: Metrics = _
+  var raftIoThread: RaftIoThread = _
```

Comment:
Why do we need 2 threads?

Reply:
It's more conventional to let the main thread be idle.